data.go

```diff
@@ -174,49 +183,70 @@ func (d *Data) Datasource(alias string, args ...string) map[string]interface{} {
 	if err != nil {
 		log.Fatalf("Couldn't read datasource '%s': %s", alias, err)
 	}
-	if source.Type == "application/json" {
-		ty := &TypeConv{}
-		return ty.JSON(string(b))
-	}
-	if source.Type == "application/yaml" {
-		ty := &TypeConv{}
-		return ty.YAML(string(b))
-	}
-	log.Fatalf("Datasources of type %s not yet supported", source.Type)
-	return nil
+	return b
 }
 
 // ReadSource -
-func (d *Data) ReadSource(fs vfs.Filesystem, source *Source, args ...string) ([]byte, error) {
-	if d.cache == nil {
-		d.cache = make(map[string][]byte)
-	}
-	cacheKey := source.Alias
-	for _, v := range args {
-		cacheKey += v
-	}
-	cached, ok := d.cache[cacheKey]
-	if ok {
+func (d *Data) ReadSource(fs vfs.Filesystem, source *Source, args ...string) (map[string]interface{}, error) {
```

Comment:
this function is public API - can you find a way to accomplish this without changing the signature?